### PR TITLE
8257872: UL: -Xlog does not check number of options

### PR DESCRIPTION
**Pull request by jyukutyo**
In Unified Logging options can be specified up to four (`-Xlog[:[selections][:[output][:[decorators][:output-options]]]]`).
But it’s possible to specify options with more colons like `java -Xlog:::::`.
It’s better to show a message when more options are specified than maximum number.

By this PR a warning message is outputted when options are specified more than the max number.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257872](https://bugs.openjdk.java.net/browse/JDK-8257872): UL: -Xlog does not check number of options

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)
 * [Yasumasa Suenaga](https://openjdk.java.net/census#ysuenaga) (@YaSuenag - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1685/head:pull/1685`
`$ git checkout pull/1685`

### CODE REVIEW

File: src/hotspot/share/logging/logConfiguration.cpp [hunk 1]
```diff
@@ -362,10 +362,15 @@ bool LogConfiguration::parse_command_line_arguments(const char* opts) {
       *next = '\0';
       str = next + 1;
     } else {
+      str = NULL;
       break;
     }
   }
 
+  if (str != NULL) {
+    log_warning(logging)("Options are specified more than the max number in -Xlog.");
```
Comment (dholmes-ora) [inline src/hotspot/share/logging/logConfiguration.cpp:371]: Might I suggest:
`log_warning(logging)("Ignoring excess -Xlog options: \"%s\"", str);`

Comment (jyukutyo) [reply src/hotspot/share/logging/logConfiguration.cpp]: Thank you for your suggestion. This message is clear for everyone. I changed the warning message.

**Comment by iklam** [comment]
Should this be a warning or an error? The behavior seems inconsistent:

```
$ java -Xlog:gc=error:::foo -version
[0.000s][warning][logging] Output options for existing outputs are ignored.
java version "16-internal" 2021-03-16
Java(TM) SE Runtime Environment (build 16-internal+0-adhoc.iklam.open)
Java HotSpot(TM) 64-Bit Server VM (build 16-internal+0-adhoc.iklam.open, mixed mode, sharing)

$ java -Xlog:gc=error::foo -version
[0.000s][error][logging] Invalid decorator 'foo'.
Invalid -Xlog option '-Xlog:gc=error::foo', see error log for details.
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

**Comment by jyukutyo** [comment]
In this PR I chose a warning level to ignore the invalid option and run a application. It isn't serious thing and some people might run with invalid option already. 

The fourth option is for output-option and can be described relatively free. So I suspect that it is ignored even when it is invalid.

**Comment by dholmes-ora** [comment]
I think a warning is okay in this context.

**Review by dholmes-ora — APPROVED** [review]

**Review by iklam — APPROVED** [review]
Agree that warning is OK.

**Review by YaSuenag — APPROVED** [review]
Looks good! I will sponsor you.

**Pull request closed without merging**